src/saving/SaveList.cs

```diff
@@ -53,12 +58,17 @@ public override void _Ready()
 
     public override void _Process(float delta)
     {
-        if (AutoRefreshOnFirstVisible && !refreshedAtLeastOnce && IsVisibleInTree())
+        if ((AutoRefreshOnFirstVisible && !refreshedAtLeastOnce && IsVisibleInTree()) ||
+            (AutoRefreshOnBecomingVisible && !wasVisible && IsVisibleInTree()))
```

Comment:
This should be reordered like this:
`bool isCurrentlyVisible = IsVisibleInTree();`

`if(isCurrentlyVisible && ((AutoRefreshOnFirstVisible && !refreshedAtLeastOnce) || (AutoRefreshOnBecomingVisible && !wasVisible)))`